ui-lib/src/main/java/co/electriccoin/zcash/ui/screen/receive/view/ReceiveView.kt

```diff
@@ -170,73 +162,139 @@ private fun ReceiveContents(
     versionInfo: VersionInfo,
     modifier: Modifier = Modifier,
 ) {
-    Column(
-        modifier =
-            Modifier
-                .fillMaxHeight()
-                .verticalScroll(rememberScrollState())
-                .then(modifier),
-        horizontalAlignment = Alignment.CenterHorizontally
-    ) {
-        if (screenBrightnessState == ScreenBrightnessState.FULL) {
-            BrightenScreen()
-            DisableScreenTimeout()
+    if (screenBrightnessState == ScreenBrightnessState.FULL) {
+        BrightenScreen()
+        DisableScreenTimeout()
+    }
+
+    val state by remember {
+        derivedStateOf {
+            listOfNotNull(
+                walletAddress.unified,
+                walletAddress.sapling.takeIf { versionInfo.isTestnet },
+                walletAddress.transparent,
+            )
         }
+    }
 
-        Spacer(modifier = Modifier.height(ZcashTheme.dimens.spacingSmall))
+    val pagerState = rememberPagerState { state.size }
 
-        Address(
-            walletAddress = walletAddress.unified,
-            onAddressCopyToClipboard = onAddressCopyToClipboard,
-            onQrImageShare = onQrImageShare,
+    Column(
+        modifier = modifier,
+    ) {
+        AddressTabs(
+            modifier = Modifier.fillMaxWidth(),
+            pagerState = pagerState,
+            tabs = state,
         )
-
-        if (versionInfo.isTestnet) {
-            Spacer(modifier = Modifier.height(ZcashTheme.dimens.spacingHuge))
-
-            Address(
-                walletAddress = walletAddress.sapling,
+        HorizontalPager(
+            modifier = Modifier.fillMaxSize(),
+            state = pagerState,
+            userScrollEnabled = false
+        ) { index ->
+            AddressPage(
+                modifier = Modifier.fillMaxSize(),
+                walletAddress = state[index],
                 onAddressCopyToClipboard = onAddressCopyToClipboard,
                 onQrImageShare = onQrImageShare,
             )
         }
+    }
+}
 
-        Spacer(modifier = Modifier.height(ZcashTheme.dimens.spacingHuge))
-
-        Address(
-            walletAddress = walletAddress.transparent,
-            onAddressCopyToClipboard = onAddressCopyToClipboard,
-            onQrImageShare = onQrImageShare,
-        )
+@OptIn(ExperimentalFoundationApi::class)
+@Composable
+private fun AddressTabs(
```

Comment:
I'm thinking about moving this UI component to the design-lib module so that it is available for possible other usage, the same as we do with the buttons, for example. What do you think?

Reply:
Yes, totally doable. Judging from figma there are several occurrences of these tabs so yes they should be reused